scripts/gitlint/rules.py

```diff
@@ -70,7 +70,8 @@ def validate(self, line, _commit):
         summary = match.group(2)
         self.log.debug(f"\nComponent: {component}\nSummary: {summary}")
 
-        if component not in valid_components:
+        if not component.startswith(valid_components):
```

Comment:
That is correct, I thought second part can be user defined, which is not validated by gitlint rule. For example, here, in #5846 commit 5f1791e4387cb52ec99709af3294bf03d7c3de00 second part of component is not in valid components list.

Also, currently there is no rule how to  separate both parts, some commits use comma others use colon.

Reply:
There may be inconsistent commits in our past, there is no need to worry about those. Let's just work to make things nicer going forward (and remember this is to advise the reviewers not be a required check since the link length rule will be broken by any clippy fix or build error patch that contains a log.)

Reply:
The component prefix should only include "valid" components that are separated by comma and space. The rest of the title is simply "Summary of Change" that may include submodule, like this commit https://github.com/cloud-hypervisor/cloud-hypervisor/commit/9c994f882a359ce0283335bf8fd951772f35b60d.

Reply:
@likebreath  I've update PR in accordance to your review

Valid title format:
component1[, component2, componentN]: submodule: summary

Title should have at least one component
Components are separated by comma+space: ", "
Components are validated to be in valid_components
Components list is ended by a colon
Submodules are not validated